tools/pubsys/src/aws/ami/mod.rs

```diff
@@ -119,77 +118,58 @@ async fn _run(args: &Args, ami_args: &AmiArgs) -> Result<HashMap<String, Image>>
     // We register in this base region first, then copy from there to any other regions.
     let base_region = regions.remove(0);
 
+    let arch = ami_args.arch.as_str().to_owned();
```

Comment:
nit: I would drop the `.to_owned()` here and do that for the places that need it.
```suggestion
    let arch = ami_args.arch.as_str();
```

Then when you're passing the `arch` to the error enum variants, you can just for example do:
```
...
    .context(error::GetAmiIdSnafu {
        name: &ami_args.name,
        arch,
        region: &base_region,
    })?;
```

Same comment applies elsewhere where you do similar things with `let arch`.

Reply:
I was running into borrow issues earlier, but it looks like they aren't there now.